src/react-components/ui-root.js

```diff
@@ -718,13 +723,16 @@ class UIRoot extends Component {
   };
 
   fetchMicDevices = () => {
+    console.log("Available input devices: ");
     return new Promise(resolve => {
       navigator.mediaDevices.enumerateDevices().then(mediaDevices => {
+        const inputDevices = mediaDevices
+          .filter(d => d.kind === "audioinput")
+          .map(d => ({ deviceId: d.deviceId, label: d.label }));
+        inputDevices.forEach(d => console.log(`\t${d.label} (${d.deviceId})`));
         this.setState(
           {
-            micDevices: mediaDevices
-              .filter(d => d.kind === "audioinput")
-              .map(d => ({ deviceId: d.deviceId, label: d.label }))
+            micDevices: inputDevices
```

Comment:
Can drop `inputDevices` inline the `filter`+`map` as it was before since you no longer need the `console.log`s above

Reply:
Updated